swarm/storage/mru/resourcemetadata.go

```diff
@@ -0,0 +1,64 @@
+package mru
+
+import (
+	"encoding/binary"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/swarm/storage"
+)
+
+type resourceMetadata struct {
+	startTime uint64
+	frequency uint64
+	name      string
+	ownerAddr common.Address
+}
+
+func (r *resourceMetadata) UnmarshalBinary(chunkData []byte) error {
+	metadataChunkLength := binary.LittleEndian.Uint16(chunkData[2:6])
```

Comment:
Encoding is totally refactored in #704, so better discuss encoding there, since it is a lot easier to visualize how messages are encoded and we can change it very quickly if we don't agree how it works.